### PR TITLE
Core home migration

### PR DESCRIPTION
**Pull request by rohitchaskar-rishabh**

### CODE REVIEW

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export function matomoActivityIndicatorAdapter(piwik: any) {
+    return {
+        restrict: 'A',
+        transclude: true,
+        scope: {
+            loading: '=',
+            loadingMessage: '=?'
+        },
+        template: `<matomo-activity-indicator-downgrade [scope]="scope" [loadingMessage]="loadingMessage" [loading]="loading" ng-show="loading" class="loadingPiwik">
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts:13]: Why do we need scope here ?

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'matomo-activity-indicator',
+  template: `<div ng-show="loading" class="loadingPiwik">
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts:5]: please update `ng-show` to *ngIf.

File: plugins/CoreHome/angular/src/lib/common/directives/attributes.directive.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    if (changes.piwikAttributes.currentValue && changes.piwikAttributes.currentValue !== '[object Object]') {
+      this.applyAttributes(changes.piwikAttributes.currentValue);
+    }
+  }
+
+  applyAttributes(attributes: any) {
+    if (typeof attributes === 'object') {
+      Object.entries(attributes).forEach(
+        ([key]) => {
+          console.log(key, attributes[key]);
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/common/directives/attributes.directive.ts:24]: Remove unnecessary consoles.

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts (hunk 2 above)
Comment (rohitchaskar-rishabh) [reply plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts]: Fixed.

File: plugins/CoreHome/angular/src/lib/common/directives/attributes.directive.ts (hunk 3 above)
Comment (rohitchaskar-rishabh) [reply plugins/CoreHome/angular/src/lib/common/directives/attributes.directive.ts]: Removed.

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts (hunk 1 above)
Comment (rohitchaskar-rishabh) [reply plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts:13]: We don't need this, removing the scope.

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+              <img src="plugins/Morpheus/images/loading-blue.gif" alt="" /> <span>{{ loadingmessage }}</span>
+              </div>
+  `,
+  styles: [
+  ]
+})
+export class ActivityIndicatorComponent implements OnInit {
+  @Input() loading = false;
+  @Input() loadingmessage = '';
+  @Input() scope:any;
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.component.ts:15]: can you remove `scope` it form here as well ?

File: plugins/CoreHome/angular/src/lib/enriched-headline/enriched-headline.component.ts [hunk 5]
```diff
@@ -77,13 +77,16 @@ declare var $: any;
...
     constructor(private componentElement: ElementRef) {}
-
-    @Input() helpUrl: string = '';
-    @Input() editUrl: string = '';
-    @Input() reportGenerated?: string;
-    @Input() featureName: string = '';
-    @Input() inlineHelp?: string = '';
-    @Input() showReportGenerated: boolean = false;
+/**
+ * All the varibale are lowecase due to issue with angular js attributes not allowing
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/enriched-headline/enriched-headline.component.ts:81]: Great!

File: plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+export function matomoActivityIndicatorAdapter(piwik: any) {
+    return {
+        restrict: 'A',
+        transclude: true,
+        scope: {
+            loading: '=',
+            loadingMessage: '=?'
+        },
+        template: `<matomo-activity-indicator-downgrade [loadingMessage]="loadingMessage" [loading]="loading" ng-show="loading" class="loadingPiwik">
```
Comment (sagarvaishnav) [inline plugins/CoreHome/angular/src/lib/activity-indicator/activity-indicator.adapter.ts:13]: I mean, wherever you find `ng-show`, it should be `*ngIf`.

**Review by sagarvaishnav — APPROVED** [review]

**Pull request merged**